src/main/java/com/workshop/demo/service/impl/RestaurantServiceImpl.java

```diff
@@ -44,13 +47,18 @@ public Integer getScore(RestaurantRequest restaurantRequest) {
 
     @Override
     public Restaurant addRestaurant(RestaurantRequest restaurantRequest, User user) {
-        Restaurant restaurant = restaurantRepository.findByName(restaurantRequest.getRestaurantName())
-                .orElseThrow(() -> new ResourceNotFoundException(THIS_RESTAURANT, ID_STR,
-                        restaurantRequest.getRestaurantName()));
-        restaurant.setCreatedAt(restaurantRequest.getCreatedAt());
+        Optional<Restaurant> restaurantOptional = restaurantRepository
+                .findByName(restaurantRequest.getRestaurantName());
+        if (restaurantOptional.isPresent()) {
+            throw new BadRequestException("Has existing restaurant with this name");
+        }
+        Restaurant restaurant = new Restaurant();
+        final SimpleDateFormat formatter = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+        Date d1 = new Date(System.currentTimeMillis());
+        restaurant.setCreatedAt(formatter.format(d1));
         restaurant.setCreatedBy(restaurantRequest.getCreatedBy());
```

Comment:
CreatedBy should use the current user (user who's in session)'s id. Now that we changed `User user` to `UserPrincipal user`, that will be the user in session.
Then we can set createdBy as `restaurant.setCreatedBy(user.getId())`